apb.yml

```diff
@@ -16,8 +16,16 @@ plans:
   - name: default
     description: This plan sets up the Sync Application
     free: True
-    metadata: {}
+    metadata: 
+      mobileclient_bind_parameters_data:
+        - '{}'
     parameters:
+      - name: SYNC_APP_NAME
+        description: The name of your sync application.
+        type: string
+        pattern: "^[a-zA-Z_][a-zA-Z0-9_]*$"
```

Comment:
Not sure about the name. I named it `vsazel_sync_app` and it ended with following exception:
`fatal: [localhost]: FAILED! => {"changed": false, "error": 422, "msg": "Failed to create object: DeploymentConfig \"sync-app-vsazel_sync_app\" is invalid: [metadata.name: Invalid value: \"sync-app-vsazel_sync_app\": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.template.spec.containers[0].name: Invalid value: \"sync-app-vsazel_sync_app\": a DNS-1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')]"}` 
Maybe replacing undescore with minus sign?

Reply:
I think the underscore just has to go, there are places where a minus sign won't work.  Alternatively I can replace the _ with a minus where a minus is needed.

@wei-lee thoughts?

Reply:
I am ok with remove underscore support. Keep it the same ask the validation used by k8s.